usaspending_api/etl/tests/integration/test_load_transactions_in_delta.py

```diff
@@ -0,0 +1,1264 @@
+
+    @staticmethod
+    def happy_verify_award_ids(spark):
+        # Verify award_id_lookup table
+        query = "SELECT * FROM int.award_id_lookup ORDER BY id, transaction_unique_id"
+        delta_data = [row.asDict() for row in spark.sql(query).collect()]
+        assert equal_datasets(TestInitialRun.expected_award_id_lookup, delta_data, "")
+
+        # Verify max award id
+        with connection.cursor() as cursor:
+            cursor.execute("SELECT nextval('award_id_seq')")
+            max_award_id = cursor.fetchone()[0] - 1
+        assert max_award_id == 6
+
+        # Verify last load date (NOTE: get_last_load_date actually returns a datetime)
+        last_load_dt = get_last_load_date("award_id_lookup")
+        assert date(last_load_dt.year, last_load_dt.month, last_load_dt.day) == date.today()
+
+    @staticmethod
+    def happy_verify(spark):
+        TestInitialRun.happy_verify_transaction_ids(spark)
+        TestInitialRun.happy_verify_award_ids(spark)
+
+    @mark.django_db(transaction=True)
+    def test_happy(self, spark, s3_unittest_data_bucket, hive_unittest_metastore_db, populate_initial_postgres_data):
+        TestInitialRun.initial_run(spark, s3_unittest_data_bucket)
+        TestInitialRun.happy_verify(spark)
+
+    @mark.django_db(transaction=True)
+    def test_run_twice(
+        self, spark, s3_unittest_data_bucket, hive_unittest_metastore_db, populate_initial_postgres_data
+    ):
+        # Verify that calling initial_run twice yields the same results as calling it once.
+        TestInitialRun.initial_run(spark, s3_unittest_data_bucket)
+        call_command("load_transactions_in_delta", "--etl-level", "initial_run")
+        TestInitialRun.happy_verify(spark)
+
+
+class TestTransactionIdLookup:
+    @mark.django_db(transaction=True)
+    def test_no_initial_run(
+        self, spark, s3_unittest_data_bucket, hive_unittest_metastore_db, populate_initial_postgres_data
+    ):
+        load_initial_delta_tables(spark, s3_unittest_data_bucket)
+
+        with raises(pyspark.sql.utils.AnalysisException, match="Table or view not found: int.transaction_id_lookup"):
+            call_command("load_transactions_in_delta", "--etl-level", "transaction_id_lookup")
+
+    @mark.django_db(transaction=True)
+    def test_no_deletes_or_inserts(
+        self, spark, s3_unittest_data_bucket, hive_unittest_metastore_db, populate_initial_postgres_data
+    ):
+        # With no deletes or inserts, the transaction_lookup_id table should be the same as after
+        # the initial run
+        TestInitialRun.initial_run(spark, s3_unittest_data_bucket)
+        call_command("load_transactions_in_delta", "--etl-level", "transaction_id_lookup")
+        TestInitialRun.happy_verify_transaction_ids(spark)
+
+    @mark.django_db(transaction=True)
+    def test_only_deletes_only_fabs(
+        self, spark, s3_unittest_data_bucket, hive_unittest_metastore_db, populate_initial_postgres_data
+    ):
```

Comment:
Can some of the combinations of tests in this class be combined to cutback on Spark jobs as well? Same for `TestAwardIdLookup` below.